pkgs/build-support/dotnet/build-dotnet-module/hooks/dotnet-configure-hook.sh

```diff
@@ -13,6 +13,11 @@ dotnetConfigureHook() {
         parallelFlag="--disable-parallel"
     fi
 
+    # Required for cross compiling, but breaks at least one package
+    if ! [ "${dontDotnetGlobalizationInvariantFlag}" ]; then
+        dotnetFlags+=("-p:InvariantGlobalization=true")
```

Comment:
Since its used only on one package, isnt it better to set dotnetFlags there instead of adding another config option to buildDotnetModule?

Reply:
The flag is used for all packages except the ones enabling this flag though, not sure what a better solution would be.

Reply:
Oh, right, I misunderstood that part.

Still, disabling globalization has some implications for application behaviour (https://github.com/dotnet/runtime/blob/main/docs/design/features/globalization-invariant-mode.md). I'm not sure enabling it globally is a good idea, since it may introduce application bugs specific to NixOS, that may be undetected if the application tests don't cover invariant mode cases.

I'm not sure what a better alternative is. Is there a way to add some sort of warning during the build or a documentation note? That way at least someone packaging a new application will have some information if they run into globalization-specific bugs.

Reply:
> Still, disabling globalization has some implications for application behaviour (https://github.com/dotnet/runtime/blob/main/docs/design/features/globalization-invariant-mode.md). I'm not sure enabling it globally is a good idea, since it may introduce application bugs specific to NixOS, that may be undetected if the application tests don't cover invariant mode cases.

That's a good point, I hadn't considered that before. After looking into this further I found that we can just use the `icu` package used in `build-dotnet.nix` and wrap it into `LD_LIBRARY_PATH`, that works on both the natively and cross compiled binaries. With that we don't need this flag anymore :+1: